bundles/org.openhab.binding.dirigera/src/main/java/org/openhab/binding/dirigera/internal/handler/light/TemperatureLightHandler.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) 2010-2025 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.dirigera.internal.handler.light;
+
+import static org.openhab.binding.dirigera.internal.Constants.CHANNEL_LIGHT_TEMPERATURE;
+
+import java.util.Iterator;
+import java.util.Map;
+import java.util.TreeMap;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.json.JSONObject;
+import org.openhab.binding.dirigera.internal.interfaces.Model;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.PercentType;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.types.Command;
+
+/**
+ * {@link TemperatureLightHandler} for lights with brightness and color temperature
+ *
+ * @author Bernd Weymann - Initial contribution
+ */
+@NonNullByDefault
+public class TemperatureLightHandler extends DimmableLightHandler {
+    // default values of "standard IKEA lamps" from JSON
+    private int colorTemperatureMax = 2202;
```

Comment:
Min and Max are reversed.
Also (I did not check it) but you should to add these constants to the channel-type xml state description min/max values; plus unit=Kelvin.

Reply:
And BTW you may find that different lamp models have different color temperature ranges. Perhaps you can read those min/max values from the bridge (as properties) on start up, and then have a dynamic state description provider to give the actual values to OH UI at runtime. => See my code for Hue API v2 here..

https://github.com/openhab/openhab-addons/blob/main/bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java

Reply:
Done